drydock_provisioner/config.py

```diff
@@ -12,33 +12,57 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+from oslo_config import cfg
 
-#
-# Read application configuration
-#
+class DrydockConfig(object):
+    """
+    Initialize all the core options
+    """
+    # Core/General options
+    logging_options = [
+        cfg.StrOpt('log_level', default='INFO', help='Global log level for Drydock'),
+        cfg.StrOpt('global_logger_name', default='drydock', help='Logger name for the top-level logger'),
+        cfg.StrOpt('oobdriver_logger_name', default='${global_logger_name}.oobdriver'),
+        cfg.StrOpt('nodedriver_logger_name', default='${global_logger_name}.nodedriver'),
+        cfg.StrOpt('control_logger_name', default='${global_logger_name}.control'),
+    ]
 
-# configuration map with defaults
+    # API Authentication options
+    auth_options = [
+        cfg.StrOpt('admin_token', default='bigboss', help='X-Auth-Token value to bypass backend authentication'),
+        cfg.BoolOpt('bypass_enabled', default=False, help='Can backend authentication be bypassed?'),
+    ]
+
+    # Enabled plugins
+    plugin_options = [
+        cfg.MultiStrOpt('ingester',
+                        default=['drydock_provisioner.ingester.plugins.yaml.YamlIngester'],
+                        help='Module path string of a input ingester to enable'),
+        cfg.MultiStrOpt('oob_driver',
+                        default=['drydock_provisioner.drivers.oob.pyghmi_driver.PyghmiDriver'],
+                        help='Module path string of a OOB driver to enable'),
+        cfg.StrOpt('node_driver',
+                    default='drydock_provisioner.drivers.node.maasdriver.driver.MaasNodeDriver',
+                    help='Module path string of the Node driver to enable'),
+        cfg.StrOpt('network_driver',
+                    default=None, help='Module path string of the Network driver to enable'),
```

Comment:
Is the network_driver being populated to a non-None value anywhere or is this just not being used in the code base yet? If so, I think adding a TODO above this would be good.